test.js

```diff
@@ -0,0 +1,204 @@
+    };
+  }
+  else {
+    expected = testCase.out;
+  }
+
+  for( var ind = 0; ind < apiResults.length; ind++ ){
+    var result = apiResults[ ind ];
+    if( equalProperties( expected, result.properties ) ){
+      var success = ( ind + 1 ) <= priorityThresh;
+      return ( success ) ?
+        { result: 'pass' } :
+        {
+          result: 'fail',
+          msg: util.format( 'Result found, but not in top %s.', priorityThresh )
+        }
+    }
+  }
+
+  return {
+    result: 'fail',
+    msg: 'No result found.'
+  }
+}
+
+/**
+ * Execute all the tests in a test-suite file with `evalTest()`, and pass an
+ * object containing the results to `cb()`. `apiUrl` contains the URL of the
+ * Pelias API to query.
+ */
+function execTestSuite( apiUrl, testSuite, cb ){
+  var testResults = {
+    stats: {
+      pass: 0,
+      fail: 0,
+      placeholder: 0,
+      timeTaken: null,
+      name: testSuite.name
+    },
+    results: []
+  };
+
+  var startTime = new Date().getTime();
+  testSuite.tests.forEach( function ( testCase ){
+    supertest( apiUrl )
+      .get( '/search?' + querystring.stringify( testCase.in ) )
+      .expect( 'Content-Type', /json/ )
+      .expect( 200 )
+      .end( function ( err, res ) {
+        if( err ){
+          throw err;
+        }
+
+        var priority = ( 'priorityThresh' in res ) ?
+          result.priorityThresh :
+          testSuite.priorityThresh;
+
+        var results = evalTest( priority, testCase, res.body.features );
+        results.testCase = testCase;
+        testResults.stats[ results.result ]++;
+        testResults.results.push( results );
+
+        if( testResults.results.length === testSuite.tests.length ){
+          testResults.stats.timeTaken = new Date().getTime() - startTime;
+          testResults.results.sort( function ( a, b ){
+            return (a.testCase.id > b.testCase.id) ? 1 : -1;
+          });
+          cb( testResults );
+        }
+      });
+  });
+}
+
+/**
+ * Asynchronously execute the given `testSuites` against the Pelias API running
+ * at `apiUrl`, and pass the results to the `outputGenerator` function.
+ */
+function execTestSuites( apiUrl, testSuites, outputGenerator ){
+  var suiteResults = {
+    stats: {
+      pass: 0,
+      fail: 0,
+      placeholder: 0,
+      timeTaken: 0,
+      url: apiUrl
+    },
+    results: []
+  };
+
+  testSuites.map( function ( suite ){
+    execTestSuite( apiUrl, suite, function ( testResults ){
+      suiteResults.results.push( testResults );
+
+      [ 'pass', 'fail', 'placeholder', 'timeTaken' ].forEach( function ( propName ){
+        suiteResults.stats[ propName ] += testResults.stats[ propName ];
+      });
+
+      if( suiteResults.results.length === testSuites.length ){
+        outputGenerator( suiteResults );
+      }
+    });
+  });
+}
+
+/**
+ * URLs for the various Pelias APIs out in the wild. Can be specified as a
+ * command-line argument (see `runTests()`).
+ */
+var PELIAS_ENDPOINTS = {
```

Comment:
these should live in a config file somewhere, and you should be able to specify a different url as well.